src/libcore/iter.rs

```diff
@@ -149,8 +156,10 @@ pub trait Iterator {
     #[stable(feature = "rust1", since = "1.0.0")]
     fn nth(&mut self, mut n: usize) -> Option<Self::Item> where Self: Sized {
         for x in self.by_ref() {
-            if n == 0 { return Some(x) }
-            n -= 1;
+            n = match n.checked_sub(1) {
+                Some(nn) => nn,
+                None => return Some(x),
+            }
```

Comment:
At least for me it looks like the checked variant is a bit slower?

```
test subtract_one        ... bench:   1873929 ns/iter (+/- 119790)
test subtract_one_option ... bench:   2078510 ns/iter (+/- 96528)
````

What were the results you were seeing? I compiled with `rustc foo.rs -Clto -Copt-level=3 --test`.